### ISSUE TITLE
 Predict on an image and save the result

### ISSUE THREAD

**Issue by franjamonga** [NLPPower/Aerial-Unet#2]
Hello ,

Thank you very much for your help. I have one last question, what if I wanted to predict on an image and save the prediction that I should make? That is, in the last part of the code I would like to know how to modify it.

Thank you very much for your attention and I await your response.

**Comment by NLPPower** [comment]
for DatasetV1Adapter' object has no attribute 'ndim you need to pay attention to the difference between callback from TF and from keras . They both have callback functions. To save prediction use fig.savefig('path/filename.png')

**Comment by franjamonga** [comment]
@NLPPower  Thank you , 

I have already solved that issue. In the last question, I refer to this part that takes the test data from the dataset.

data_aug_iter = test_ds.make_one_shot_iterator()
next_element = data_aug_iter.get_next()

for i in range(40):
  plt.figure(figsize=(18, 18))
  batch_of_imgs, label = tf.keras.backend.get_session().run(next_element)
  img = batch_of_imgs[0]
  predicted_label = model.predict(batch_of_imgs)[0]

  plt.subplot(1, 3,1)
  plt.imshow(img)
  plt.title("Input image")
  
  plt.subplot(1, 3,2)
  plt.imshow(label[0, :, :, 0])
  plt.title("Actual Mask")
  plt.subplot(1,3,3)
  plt.imshow(predicted_label[:, :, 0])
  plt.title("Predicted Mask")
  plt.show()

I would like to open a directory of my own images in png format to be able to predict on them with the trained weights that the model generates and save each image prediction.

thank you very much for your attention. I hope you can solve my question

**Comment by NLPPower** [comment]
You can't feed the images to the model without pre-processing it as tensor data. Repeat the step I used for pre-processing data without using data augmentation functions. s. Using this function:
train_dataset = get_data(X_train,Y_train,preproc_fn=train_preprocessing,batch_size=batch_size) to generate this dataset where X_train or X_predict in your list is the list of png images.
I think I will reconstruct the whole code soon on Tensorflow instead of Keras so you may expect a complete change of the code.

**Comment by franjamonga** [comment]
Thank you very much for the information, it is a very complete code. In this case that tells me my images to predict in png would be the X train in this case X predict, the only doubt is that these images that I would like to predict have no associated label so I imagine that Y train in this case Y predict I shouldn't put it. It would be as follows:

train_dataset = get_data(X_train,preproc_fn=train_preprocessing,batch_size=batch_size) 

In this case x train would be my set of images to predict X predict. I would like to know if put like this would be correct. I will be looking forward to seeing your new code. Receive a coordial greeting and thank you very much again for the help.
